api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/AnnotationBuilder.java

```diff
@@ -0,0 +1,764 @@
+     *     or {@link jakarta.enterprise.lang.model.types.ClassType ClassType}.</li>
+     * </ul>
+     *
+     * @param value the class value, must not be {@code null}
+     * @return this {@code AnnotationBuilder}
+     * @throws IllegalArgumentException if given type is invalid, as described above
+     */
+    default AnnotationBuilder value(Type value) {
+        return member(AnnotationInfo.MEMBER_VALUE, value);
+    }
+
+    /**
+     * Adds a class array-valued annotation member called {@code value}.
+     * The {@code values} parameter may only contain:
+     * <ul>
+     *     <li>{@link jakarta.enterprise.lang.model.types.VoidType VoidType};</li>
+     *     <li>{@link jakarta.enterprise.lang.model.types.PrimitiveType PrimitiveType};</li>
+     *     <li>{@link jakarta.enterprise.lang.model.types.ClassType ClassType};</li>
+     *     <li>{@link jakarta.enterprise.lang.model.types.ArrayType ArrayType}
+     *     whose component type is either {@link jakarta.enterprise.lang.model.types.PrimitiveType PrimitiveType}
+     *     or {@link jakarta.enterprise.lang.model.types.ClassType ClassType}.</li>
+     * </ul>
+     *
+     * @param values the class array, must not be {@code null} or contain {@code null}
+     * @return this {@code AnnotationBuilder}
+     * @throws IllegalArgumentException if any given type is invalid, as described above
+     */
+    default AnnotationBuilder value(Type... values) {
+        return member(AnnotationInfo.MEMBER_VALUE, values);
+    }
+
+    /**
+     * Adds an annotation-valued annotation member called {@code value}.
+     *
+     * @param value the annotation value, must not be {@code null}
+     * @return this {@code AnnotationBuilder}
+     */
+    default AnnotationBuilder value(AnnotationInfo<?> value) {
+        return member(AnnotationInfo.MEMBER_VALUE, value);
+    }
+
+    /**
+     * Adds an annotation array-valued annotation member called {@code value}.
+     *
+     * @param values the annotation array, must not be {@code null} or contain {@code null}
+     * @return this {@code AnnotationBuilder}
+     */
+    default AnnotationBuilder value(AnnotationInfo<?>... values) {
+        return member(AnnotationInfo.MEMBER_VALUE, values);
+    }
+
+    /**
+     * Adds an annotation-valued annotation member called {@code value}.
+     *
+     * @param value the annotation value, must not be {@code null}
+     * @return this {@code AnnotationBuilder}
+     */
+    default AnnotationBuilder value(Annotation value) {
+        return member(AnnotationInfo.MEMBER_VALUE, value);
+    }
+
+    /**
+     * Adds an annotation array-valued annotation member called {@code value}.
+     *
+     * @param values the annotation array, must not be {@code null} or contain {@code null}
+     * @return this {@code AnnotationBuilder}
+     */
+    default AnnotationBuilder value(Annotation... values) {
+        return member(AnnotationInfo.MEMBER_VALUE, values);
+    }
```

Comment:
These methods will require implementors to transform `Annotation` instances to `AnnotationInfo` internally presumably via reflection, which is probably doable but more overhead

Reply:
I added this to let users use the existing `AnnotationLiteral` subclasses. Indeed it requires reflection, but I found it's not that hard to implement -- here's how I transform an `Annotation` to Jandex `AnnotationInstance`: https://github.com/Ladicek/quarkus-fork/blob/experiment-cdi-lite-ext-v5/independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/cdi/lite/ext/AnnotationsReflection.java

I won't insist on keeping it, but I do think it's convenient.

Reply:
Ok it is probably fine.

Reply:
If you later find out it's problematic, we can always remove it. It's strictly convenience, not a requirement.